test/test_utils/ec2.py

```diff
@@ -35,11 +35,9 @@ def get_ec2_instance_type(default, processor, disable_p3dn=False):
             f"Aborting EC2 test run. Unrecognized processor type {processor}. "
             f"Please choose from {allowed_processors}"
         )
-    if default == p3dn:
-        raise RuntimeError("Default instance type should never be p3dn.24xlarge")
     instance_type = os.getenv(f"EC2_{processor.upper()}_INSTANCE_TYPE", default)
-    if instance_type == p3dn and disable_p3dn:
-        instance_type = default
+    if (default == p3dn and instance_type != p3dn) or (instance_type == p3dn and disable_p3dn):
+        return []
```

Comment:
@ChaiBapchya and I synced up offline about what the best way to include EFA tests into the PR and other contexts would be. I recommended the addition of these changes to:
a. Remove the guard rails that prevent the use of p3dn.24xlarge instance types for all tests by default
b. Add code that skips tests when the default instance type for a test is expected to be p3dn.24xlarge, but the test executor doesn't use p3dn.24xlarge instances.